### DISCUSSION TITLE
Regarding a couple failing test's from scipy-conftest.py

### DISCUSSION THREAD

**Discussion by anutosh491** [pyodide/pyodide#5080] [General]
Hi, a core developer for lfortran .... and mainly a pyodide newbie here. Landed here after going through @lesteve 's EuroScipy 2023 talk

I am interested in the topic of enabling the test suite for scipy. I started looking into 

https://github.com/pyodide/pyodide/blob/b4a6baaab4e0bdc3f6e886d062163925c1795cdc/packages/scipy/scipy-conftest.py#L90-L91

I think both these tests have something to do with the `dotu` function. Though as I always like analyzing before straightaway jumping into code.

I tried the exact same examples with `dotc` (the only difference between `dotu` and `dotc` is that `dotc` takes conjugate of the first vector under consideration)

I see that everything passes when we use `dotc`

I've been looking here but haven't found anything too obvious
https://github.com/scipy/scipy/blob/7be13dcc1b250e7c4f62e4c5d848862c7126c056/scipy/linalg/fblas_l1.pyf.src#L355-L377

Also `dot` and/or `dotc` do well (`dotc` and `dotu` almost have the same implementation) so haven't been able to point out exactly what's required to fix this.

Am I missing something ? Any guidance is welcome :)

**Comment by anutosh491** [comment]
Okay I see that there are 5 parameters in `cython_blas_signatures.txt`
```
c cdotu(int *n, c *cx, int *incx, c *cy, int *incy)
```
This doesn't agree with the docs which talk about 5 optional and 2 required parameters ( the above is missing the `offx` and `offy`)
```
cdotu(x, y[, n, offx, incx, offy, incy]) = <fortran function cdotu>
```

**Comment by agriyakhetarpal** [comment]
Hi, @anutosh491! The function signature mismatch should be possible to fix from a first look, though I don't have a lot of experience with fixing them myself as much as others do. If the test you mentioned is passing in the development version of the Pyodide console (https://pyodide.org/en/latest/console.html), could you try un-skipping it?

Edit: oops, I misread – I see that the test is for `dotu`, and not `dotc`. A patch in `packages/scipy/patches/` to fix the signature would be welcome.

**Comment by lesteve** [comment]
The best reference for debugging signature mismatch is https://pyodide.org/en/stable/development/debugging.html#debugging-runtimeerror-function-signature-mismatch. Then you need trial and error to figure out why the signature is different and how to fix it ...

For inspiration you can take a look at https://github.com/pyodide/pyodide/issues/3379. There are likely other similar issues with potentially relevant information.

**Reply by anutosh491** [reply]
Thanks for the reply.  First day looking into pyodide and it really helps :)
Yupp I was looking into it. But then i was confused as to how `dot` and `dotc` are not having the same error ... because they are framed quite similar to `dotu`.

Also what's the standard we are looking for. Is this given by the `cython_blas_signatures.txt` ? 
Cause I see all signatures are being stored in this file so I guess this is what we need to match ?

**Comment by hoodmane** [comment]
Thanks for looking into this @anutosh491! I've been confused about the cdotu situation for years but have never had the energy to look into it. It's great to have more people looking into it.

**Reply by anutosh491** [reply]
Thanks. I shall be looking into it more carefully in the upcoming days :)
And share my progress through a PR if I can get it done
